packages/transformers/webextension/src/WebExtensionTransformer.js

```diff
@@ -216,6 +216,46 @@ async function collectDependencies(
     }
     program.web_accessible_resources = war;
   }
+  if (program.declarative_net_request?.rule_resources) {
+    const rrs: {path: string, id: string, enabled: boolean}[] =
+      program.declarative_net_request?.rule_resources ?? [];
+    for (let i = 0; i < rrs.length; ++i) {
+      const resources = rrs[i];
+      const rulesPath = path.relative(assetDir, resources.path);
+      if (!(await fs.exists(rulesPath))) {
+        const pathSourceLoc = getJSONSourceLocation(
+          ptrs[`/declarative_net_request/rule_resources/${i}/path`],
+          'value',
+        );
+        throw new ThrowableDiagnostic({
+          diagnostic: [
+            {
+              message: 'Invalid Web Extension manifest',
+              origin: '@parcel/transformer-webextension',
+              codeFrames: [
+                {
+                  filePath,
+                  language: 'json',
+                  codeHighlights: [
+                    {
+                      ...pathSourceLoc,
+                      message: `Path does not exist`,
```

Comment:
We could make this include the ruleset ID, e.g. do this for `codeHighlights[0]`:

```js
{
  ...getJSONSourceLocation(
    ptrs[`/declarative_net_request/rule_resources/${i}/path`],
    'value',
  ),
  message: `Ruleset file for static ruleset ${resources.id} does not exist`
}
```

This is just to accomodate the fact that in some environments codeframes aren't fully shown, so the original message would be too vague.

Reply:
Done. I also added the invalid filepath to the error message.